ua-parser.js

```diff
@@ -21,22 +24,53 @@ class UAParser {
      * @return {String}
      */
     static getDevice(ua) {
-
         let device = _parser.setUA(ua).getDevice().type;
+        return UAParser._normalizeDeviceType(device);
+    }
 
-        if (device == null) {
+    /**
+     * @param {String} ua
+     * @return {Object}
+     */
+    static getUAData(ua) {
+        let uAData = cache.get(ua);
+        if(uAData != null) {
+            return JSON.parse(uAData);
+        }
+        let deviceData = _parser.setUA(ua).getResult();
+        uAData = {
+            deviceType: this._normalizeDeviceType(deviceData.device.type),
+            deviceModel: deviceData.device.model,
```

Comment:
Return `null` if `deviceData.device.model` is undefined.
`deviceModel: deviceData.device.model || null`

Reply:
Same applies for `deviceModel`, `browserName` and `browserVersion`